chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -1176,7 +1178,18 @@ impl Ledger {
             return Err(Error::VoteTallyProofFailed);
         }
 
-        let stake = StakeControl::new_with(&self.accounts, &self.utxos);
+        let voting_token = self
+            .votes
+            .plans
+            .lookup(tally.id())
+            // XXX: unsure about this.  This would be handled anyway inside
+            // `apply_commitee_result`, although with a different error.  But it needs to be
+            // checked here, unless this logic is moved to the VotePlanManager somehow.
```

Comment:
This part is looking good to me, the error still needs to be processed. Although if you are _absolutely_ sure that `.ok_or()` is never triggered you can replace it with `.expect()`. Let's agree on either of those and remove the comments.